### PR TITLE
chore: Add docs for introduce protocol

### LINKED ISSUE #1114: Add docs for introduce protocol
Part of #1109

### PR DESCRIPTION
**Pull request by soluchok**
* added godoc
* moved gomocks
* added example

Closes #1114 

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

**Comment by troyronda** [comment]
@soluchok Did you put a package comment on pkg/client/introduce?

(so the Synopsis column gets filled into the [GoDoc index](https://godoc.org/github.com/hyperledger/aries-framework-go))

File: pkg/client/introduce/doc.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+// Introduce protocol is responsible for the introduction between agents.
+// The protocol involves at least two participants. A maximum of three participants might be involved.
+// When creating a client a default invitation might be provided. It simplifies the usage when
+// someone wants to do introduction with your client. The simplest way to handle incoming messages (actions) is:
+//	thID, _ := action.Message.ThreadID()
+//	client.Continue(client.InvitationEnvelope(thID))
+//
+// But also there is an exception, when receiving a request.
+// After receiving a request on of the following functions must be executed.
```
Comment (troyronda) [inline pkg/client/introduce/doc.go]: on => one

File: pkg/client/introduce/example_test.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+	introduceClientProvider.EXPECT().
+		StorageProvider().
+		Return(clientStorageProvider)
+
+	return introduceClientProvider, done
+}
+
+// nolint: gocyclo
+func Example() {
+	wg := sync.WaitGroup{}
```
Comment (rolsonquadras) [inline pkg/client/introduce/example_test.go]: @troyronda @fqutishat  I remember we were discussing about avoiding WaitGroups. Does that involve test file as well ?

File: pkg/client/introduce/example_test.go [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+			MyDID:    Alice,
+			TheirDID: Bob,
+		})
+	if err != nil {
+		fmt.Println("SendProposalWithInvitation", err)
+	}
+
+	// Output:
+	// bob received https://didcomm.org/introduce/1.0/proposal
+	// alice received https://didcomm.org/introduce/1.0/response
```
Comment (rolsonquadras) [inline pkg/client/introduce/example_test.go]: Can we add an example for introduction between 3 participants as well ? ie. Alice introducing Carol to Bob. Most of the use cases of Introduce involves 3 parties.

File: pkg/client/introduce/example_test.go [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+			}
+		}
+	}()
+
+	// Send proposal
+	err = alice.SendProposalWithInvitation(&didexchange.Invitation{
+		Type: didexchange.InvitationMsgType},
+		&introduce.Recipient{
+			To: &introduce.To{
+				Name: "Bob",
```
Comment (rolsonquadras) [inline pkg/client/introduce/example_test.go]: Should this be "Bob" ? This seems like we are sending request to "Bob" (`TheirDID` field) to introduce to "Bob".

Comment (soluchok) [reply pkg/client/introduce/example_test.go]: It is just a name, It can be whatever. I may change it to Carol.

Comment (rolsonquadras) [reply pkg/client/introduce/example_test.go]: That would give more clarity.

File: pkg/client/introduce/example_test.go (hunk 3 above)
Comment (soluchok) [reply pkg/client/introduce/example_test.go]: Done!

**Comment by kdimak** [comment]
Could you add a note to docs/project_structure.md?

File: pkg/client/introduce/example_test.go (hunk 2 above)
Comment (fqutishat) [reply pkg/client/introduce/example_test.go]: @rolsonquadras  test files are okay

File: pkg/client/introduce/doc.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package introduce is responsible for the introduction between agents.
+// The protocol involves at least two participants. A maximum of three participants might be involved.
```
Comment (llorllale) [inline pkg/client/introduce/doc.go]: Change 

> A maximum of three participants might be involved.

to 

> A maximum of three participants is currently supported.

File: pkg/client/introduce/doc.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+// When creating a client a default invitation might be provided. It simplifies the usage when
+// someone wants to do introduction with your client. The simplest way to handle incoming messages (actions) is:
+//	thID, _ := action.Message.ThreadID()
+//	client.Continue(client.InvitationEnvelope(thID))
```
Comment (llorllale) [inline pkg/client/introduce/doc.go]: s/client.Continue/event.Continue/

File: pkg/client/introduce/doc.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+// - HandleRequestWithInvitation - is used when you have a public invitation.
+// A bit complicated way to handle incoming messages (actions) is:
+//	if action.Message.Type() == introduce.RequestMsgType {
+//		alice.HandleRequest(action.Message, to, recipient)
+//		OR
+//		alice.HandleRequestWithInvitation(action.Message, inv, to)
+//	}
+//	thID, _ := action.Message.ThreadID()
+//	action.Continue(alice.InvitationEnvelope(thID))
+//
```
Comment (llorllale) [inline pkg/client/introduce/doc.go]: I suggest adding a whole example that at least starts with registration of the events channel and listening for incoming events.

File: pkg/client/introduce/doc.go (hunk 5 above)
Comment (soluchok) [reply pkg/client/introduce/doc.go]: Done!

File: pkg/client/introduce/doc.go (hunk 7 above)
Comment (soluchok) [reply pkg/client/introduce/doc.go]: Done!

**Pull request merged**